### PR TITLE
B-feature/gh-411-import-requests-from-google-calendar

### PR DESCRIPTION
**Pull request by mariobasic**
Import Requests from google calendar

 - Rename Classes used for importing Users to differentiate from Calendar

 - Create job from importing, reading and writing requests from expor

GH issue: #411

### CODE REVIEW

File: src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  private Duration getDuration(CalendarImportData calendarImportData, LocalDate currentDate) {
+    LocalDateTime startDateTime = calendarImportData.getEventStart();
+    LocalDateTime endDateTime = calendarImportData.getEventEnd();
+    LocalDate startDate = startDateTime.toLocalDate();
+    LocalDate endDate = endDateTime.toLocalDate();
+
+    if (calendarImportData.isAllDayEvent()) {
+      return Duration.FULL_DAY;
+    } else if (startDate.equals(endDate)) {
+      return (startDateTime.getHour() >= NOON) ? Duration.AFTERNOON : Duration.MORNING;
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java]: Remove the parenthesis, they are useless in this case.

File: src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      return (endDateTime.getHour() <= AFTERNOON) ? Duration.MORNING : Duration.FULL_DAY;
+    } else {
+      return Duration.FULL_DAY;
+    }
+  }
+
+  private Request createRequestFromImport(CalendarImportData calendarImportData, List<Day> days) {
+    Optional<User> user = userRepository.findFirstByFullName(getUsersFullName(calendarImportData));
+    if (user.isEmpty()) {
+      return null;
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java]: It is a bad practice to return null. Could you return an empty request, or something similar?

File: src/main/java/eu/execom/hawaii/batch/CalendarDbWriter.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  @Autowired
+  public CalendarDbWriter(RequestRepository requestRepository, AllowanceService allowanceService) {
+    this.requestRepository = requestRepository;
+    this.allowanceService = allowanceService;
+  }
+
+  @Override
+  public void write(List<? extends Request> importedRequests) throws Exception {
+    importedRequests.forEach(this::createRequest);
+    importedRequests.forEach(request -> log.debug("Data saved for request '{}'.", request));
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/batch/CalendarDbWriter.java]: Why don't you put debug line also in the first loop, end of `createRequest` method?

File: src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    CALENDAR_LEAVE_TYPE.put("Approved Leave", RequestStatus.APPROVED);
+    CALENDAR_LEAVE_TYPE.put("Pending Leave", RequestStatus.PENDING);
+    CALENDAR_LEAVE_TYPE.put("Sick", RequestStatus.APPROVED);
+  }
+
+  @Override
+  public Request process(CalendarImportData calendarImportData) throws Exception {
+    List<Day> days = createDaysFromImport(calendarImportData);
+    Request request = createRequestFromImport(calendarImportData, days);
+    log.debug("Converting '{}' into request '{}' and day O '{}' number of days '{}'.", calendarImportData,
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java]: What does 0 mean in the logs?

File: src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java (hunk 2 above)
Comment (mariobasic) [reply src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java]: I read it from documentation: 
`To filter a record, one simply returns "null" from the ItemProcessor. The framework will detect that the result is "null" and avoid adding that item to the list of records delivered to the ItemWriter. As usual, an exception thrown from the ItemProcessor will result in a skip.`

Should I find a different way nonetheless?

Comment (stanislavgrujic) [reply src/main/java/eu/execom/hawaii/batch/CalendarProcessor.java]: No, if it is recommended, and needed, by the framework. Please leave it as is.

**Pull request merged**